test/e2e/agentflavor.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+package e2e
+
+import "fmt"
+
+type agentFlavor string
+
+func (af *agentFlavor) String() string {
+	return string(*af)
+}
+
+func (af *agentFlavor) Set(value string) error {
+	if len(*af) > 0 {
+		return fmt.Errorf("flavor flag already set to %s while trying to set to %s", *af, value)
+	}
+	*af = agentFlavor(value)
+	return nil
+}
+
+const (
+	agentFlavorDatadogAgent     agentFlavor = "datadog-agent"
+	agentFlavorDatadogIOTAgent  agentFlavor = "datadog-iot-agent"
+	agentFlavorDatadogDogstatsd agentFlavor = "datadog-dogstatsd"
+)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ Do we know what additional changes be needed to support `obersvability-pipelines-worker` or `vector`? It seems out of scope for these changes, but would be good to understand what future work may entail.

Reply:
This test can potentially support any value supported by `DD_AGENT_FLAVOR`.